core/integration/module_test.go

```diff
@@ -6027,6 +6027,741 @@ func (m *Test) Fn() string {
 	require.NoError(t, err)
 }
 
+func (ModuleSuite) TestContextDirectory(ctx context.Context, t *testctx.T) {
```

Comment:
nit: this may be testing too much for the SDKs. When we test multiple SDKs like this, you want to make sure you're only testing if the SDK is making the right API calls.

Then you make another test case to more fully test the engine's behavior, using `dagger call` for example, but only need the Go SDK for that. That's where you make sure all the relative and absolute paths are working as expected.

Also, forcing the request within the function and concatenating the string slices make it more complicated than it needs to be. You could just return the `Directory` directly and get the `entries` with `dagger call`.